src/main/java/com/mindera/school/spaceshiprent/service/spaceship/SpaceShipServiceImpl.java

```diff
@@ -38,25 +40,25 @@ public List<SpaceShipDetailsDto> getAllSpaceShips() {
 
     @Override
     public SpaceShipDetailsDto getSpaceShipById(Long id) {
+       Optional<SpaceshipEntity> spaceshipEntity = spaceShipRepository.findById(id);
 
-        Optional<SpaceshipEntity> spaceShipEntity = spaceShipRepository.findById(id);
+       return spaceshipEntity
+               .map(converter::convertToSpaceShipDetailsDto)
+               .orElseThrow(() -> new SpaceshipNotFoundException(String.format(ErrorMessageConstants.SPACESHIP_NOT_FOUND,id)));
```

Comment:
Although this is clean, I would prefer to do the `orElseThrow()` right in the `spaceShipRepository.findById(id)`